### PR TITLE
Add Tests for List Commands and Command Result

### PR DESCRIPTION
**Pull request by seanlaiys**
...with bug fixed for last shown list when converting sample addressbook after having addressbook.json deleted

### CODE REVIEW

File: src/test/java/seedu/address/logic/parser/ListScoreCommandParserTest.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        // whitespace only preamble
+        assertParseSuccess(parser, TUTORIALNAME_DESC_T01 + ASSESSMENT_NAME_DESC_OP1,
+                new ListScoreCommand(tutorialName, assessmentName));
+    }
+
+    @Test
+    public void parse_tutorialNameFieldMissing_failure() {
+        String expectedMessage = String.format(MESSAGE_INVALID_COMMAND_FORMAT, ListScoreCommand.MESSAGE_USAGE);
+
+        // missing day field
```
Comment (honganhcs) [inline src/test/java/seedu/address/logic/parser/ListScoreCommandParserTest.java:36]: the comment shld be missing tutorial name field!

File: src/test/java/seedu/address/logic/parser/ListScoreCommandParserTest.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        assertParseFailure(parser, ASSESSMENT_NAME_DESC_OP1, expectedMessage);
+    }
+
+    @Test
+    public void parse_assessmentNameFieldMissing_failure() {
+        String expectedMessage = String.format(MESSAGE_INVALID_COMMAND_FORMAT, ListScoreCommand.MESSAGE_USAGE);
+
+        // missing day field
```
Comment (honganhcs) [inline src/test/java/seedu/address/logic/parser/ListScoreCommandParserTest.java:44]: similar to above, with assessment name

File: src/test/java/seedu/address/logic/commands/ListScoreCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,465 @@
...
+        } catch (CommandException ce) {
+            assertEquals(expectedMessage, ce.getMessage());
+        }
+    }
+
+    @Test
+    public void execute_invalidAssessmentNameInput_commandFailure() {
+        String expectedMessage = Messages.MESSAGE_ASSESSMENT_NOT_FOUND;
+        ListScoreCommand command = new ListScoreCommand(VALID_TUTORIAL_NAME_1, VALID_ASSESSMENT_NAME_2);
+        try {
```
Comment (honganhcs) [inline src/test/java/seedu/address/logic/commands/ListScoreCommandTest.java:112]: maybe you can use assertThrows for consistency with other test classes?

File: src/test/java/seedu/address/logic/commands/ListStudentCommandTest.java [hunk 4]
```diff
@@ -0,0 +1,521 @@
...
+            assertEquals(expectedMessage, ce.getMessage());
+        }
+    }
+
+    @Test
+    public void execute_indexInput_displayListMismatchFailure() {
+        String expectedMessage = Messages.MESSAGE_INDEX_LIST_MISMATCH + ListStudentCommand.MESSAGE_INDEX_USAGE;
+        ListStudentCommand command = new ListStudentCommand(INDEX_ONE, null);
+        try {
+            ModelStubAcceptingIndex model = new ModelStubAcceptingIndex();
```
Comment (honganhcs) [inline src/test/java/seedu/address/logic/commands/ListStudentCommandTest.java:130]: same as above, should we use assertThrows instread?

File: src/test/java/seedu/address/logic/commands/ListAttendanceCommandTest.java [hunk 5]
```diff
@@ -0,0 +1,515 @@
...
+        }
+    }
+
+    @Test
+    public void execute_invalidTutorialNameInput_failure() {
+        String expectedMessage = Messages.MESSAGE_TUTORIAL_NOT_FOUND;
+        ListAttendanceCommand command =
+                new ListAttendanceCommand(T02.getTutorialName(), null, validWeek);
+        ModelStubAcceptingTutorialName model = new ModelStubAcceptingTutorialName();
+        try {
```
Comment (honganhcs) [inline src/test/java/seedu/address/logic/commands/ListAttendanceCommandTest.java:131]: same as below, should we use assertThrows instead? i think it also helps to check the type of error thrown

**Review by honganhcs — APPROVED** [review]

File: src/test/java/seedu/address/logic/parser/ListScoreCommandParserTest.java (hunk 2 above)
Comment (seanlaiys) [reply src/test/java/seedu/address/logic/parser/ListScoreCommandParserTest.java:44]: Thanks for spotting! Will make the changes

File: src/test/java/seedu/address/logic/commands/ListScoreCommandTest.java (hunk 3 above)
Comment (seanlaiys) [reply src/test/java/seedu/address/logic/commands/ListScoreCommandTest.java:112]: sounds good. Thank you!

File: src/test/java/seedu/address/logic/commands/ListStudentCommandTest.java (hunk 4 above)
Comment (seanlaiys) [reply src/test/java/seedu/address/logic/commands/ListStudentCommandTest.java:130]: Thanks! Will make the changes

File: src/test/java/seedu/address/logic/commands/ListAttendanceCommandTest.java (hunk 5 above)
Comment (seanlaiys) [reply src/test/java/seedu/address/logic/commands/ListAttendanceCommandTest.java:131]: Thanks! Will make the changes

File: src/test/java/seedu/address/logic/parser/ListScoreCommandParserTest.java (hunk 1 above)
Comment (seanlaiys) [reply src/test/java/seedu/address/logic/parser/ListScoreCommandParserTest.java:36]: Thanks for spotting! Will make the changes

**Pull request merged**